app/src/main/java/com/arctouch/codechallenge/util/schedulers/SchedulerProvider.kt

```diff
@@ -0,0 +1,18 @@
+package com.arctouch.codechallenge.util.schedulers
+
+import io.reactivex.Scheduler
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.schedulers.Schedulers
+
+
+class SchedulerProvider : BaseScheduler {
+
+    override fun io(): Scheduler {
+        return Schedulers.newThread()
```

Comment:
Sometimes I had a few problems using io() instead of newThread() because the number of the process running in the same time were too large. But I believe that in this specific case, io() would be enough.